programs/stake/src/stake_state.rs

```diff
@@ -757,31 +761,61 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
             match self.state()? {
                 StakeState::Stake(meta, mut stake) => {
                     meta.authorized.check(signers, StakeAuthorize::Staker)?;
+                    let split_rent_exempt_reserve = calculate_split_rent_exempt_reserve(
+                        meta.rent_exempt_reserve,
+                        self.data_len()? as u64,
+                        split.data_len()? as u64,
+                    );
 
                     // verify enough lamports for rent in new stake with the split
-                    if split.lamports()? + lamports < meta.rent_exempt_reserve
-                     // verify enough lamports left in previous stake and not full withdrawal
+                    if split.lamports()? + lamports < split_rent_exempt_reserve
+                    // verify enough lamports left in previous stake and not full withdrawal
                         || (lamports + meta.rent_exempt_reserve > self.lamports()? && lamports != self.lamports()?)
```

Comment:
nits: I think these conditions are bit hard to read? How about swapping this inequality around `lamports` like this?

```rust
if lamports < split_rent_excempt_reserve + lamports
    || (lamports > self.lamports()? - meta.rent_excempt_reserver && ...)
```

Reply:
✅  Reframed all those checks in terms of `lamports` (and also let cargo fmt work on those lines)